File: scala-spark/src/main/scala/com/dvtn/spark/core/operation/TransformationOperation.scala
package com.dvtn.spark.core.operation

import org.apache.spark.{SparkConf, SparkContext}

object TransformationOperation {
  def main(args: Array[String]): Unit = {
    val conf = new SparkConf().setMaster("local").setAppName("TransformationOperationTest")
    val sc = new SparkContext(conf)

    /**
     * minPartitions:
     *  指定生成的RDD的分区数,增加该参数可增加task的并行度
     */
    val lines = sc.textFile("./scala-spark/data/words", 3)


    /**
     * sample(withReplacement = , fraction = , seed = ) 抽样
     * 参数说明：
     *  withReplacement: Boolean,表示是否是放回抽样
     *  fraction: Double, 抽样比例
     *  seed: Long,种子,针对同一批数据只要定义的种子相同表示每次抽样的结果是否相同
     */
//    val result = lines.sample(true , 0.1 , 100)
//    result.foreach(println)

    /**
     * filter
     */
    lines.filter(s=>{
      s.endsWith("hello world")
    }).foreach(println)


    sc.stop()


  }

}

File: java-spark/src/main/java/com/dvtn/core/operation/TransformationOperation.java
package com.dvtn.core.operation;

import org.apache.spark.SparkConf;
import org.apache.spark.api.java.JavaRDD;
import org.apache.spark.api.java.JavaSparkContext;
import org.apache.spark.api.java.function.Function;
import org.apache.spark.api.java.function.VoidFunction;

public class TransformationOperation {
    public static void main(String[] args) {
        SparkConf conf = new SparkConf();
        conf.setMaster("local").setAppName("TransformationOperation");

        JavaSparkContext sc = new JavaSparkContext(conf);

        JavaRDD<String> lines = sc.textFile("./java-spark/data/words");



        /**
         * filter(f:T=>Bool): RDD[T] => RDD[T]
         *      过滤RDD时内容，不改变RDD的类型
         */

        //JavaRDD<String> filter = lines.filter(new Function<String, Boolean>() {
        //    @Override
        //    public Boolean call(String s) throws Exception {
        //        return s.equalsIgnoreCase("hello scala");
        //    }
        //});
        //filter.foreach(new VoidFunction<String>() {
        //    @Override
        //    public void call(String s) throws Exception {
        //        System.out.println(s);
        //    }
        //});


        /**
         * sample(boolean withReplacement, double fraction, long seed)
         * 参数
         *      withReplacement: 有无放回抽样
         *      fraction: 抽样比例
         *      seed: 抽样种子
         *
         */
        //JavaRDD<String> sample = lines.sample(true, 0.1, 100);
        //sample.foreach(new VoidFunction<String>() {
        //    @Override
        //    public void call(String s) throws Exception {
        //        System.out.println(s);
        //    }
        //});


        sc.stop();
    }
}
